contrib/platform/src/com/sun/jna/platform/win32/Winioctl.java

```diff
@@ -72,4 +95,320 @@ public STORAGE_DEVICE_NUMBER(Pointer memory) {
             return FIELDS;
         }
     }
+
+    /**
+     * Indicates the properties of a storage device or adapter to retrieve as the input buffer passed to the
+     * IOCTL_STORAGE_QUERY_PROPERTY control code.
+     */
+    class STORAGE_PROPERTY_QUERY extends Structure {
+
+        public static final List<String> FIELDS = createFieldsOrder("propertyId", "queryType", "additionalParameters");
+
+        /**
+         * Indicates whether the caller is requesting a device descriptor, an adapter descriptor, a write cache
+         * property, a device unique ID (DUID), or the device identifiers
+         * provided in the device's SCSI vital product data (VPD) page.
+         */
+        public /* STORAGE_PROPERTY_ID */ WinDef.DWORD propertyId;
+
+        /**
+         * Contains flags indicating the type of query to be performed as enumerated by the STORAGE_QUERY_TYPE
+         * enumeration.
+         */
+        public /* STORAGE_QUERY_TYPE */ WinDef.DWORD queryType;
+
+        /**
+         * Contains an array of bytes that can be used to retrieve additional parameters for specific queries.
+         */
+        public byte[] additionalParameters = new byte[1];
+
+        public static class ByReference extends STORAGE_PROPERTY_QUERY implements Structure.ByReference {
+            public ByReference() {
+            }
+        }
+
+        public STORAGE_PROPERTY_QUERY() {
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return FIELDS;
+        }
+    }
+
+    /**
+     * Contains information about the media types supported by a device.
+     */
+    class GET_MEDIA_TYPES extends Structure {
+
+        public static final List<String> FIELDS = createFieldsOrder("deviceType", "mediaInfoCount", "mediaInfo");
+
+        /**
+         * The type of device. Values from 0 through 32,767 are reserved for use by Microsoft Corporation. Values from
+         * 32,768 through 65,535 are reserved for use by other vendors.
+         */
+        public WinDef.DWORD deviceType;
+
+        /**
+         * The number of elements in the MediaInfo array.
+         */
+        public WinDef.DWORD mediaInfoCount;
+
+        /**
+         * A pointer to the first DEVICE_MEDIA_INFO structure in the array. There is one structure for each media type
+         * supported by the device.
+         */
+        public WinDef.DWORD /* DEVICE_MEDIA_INFO */ mediaInfo;
+
+        public static class ByReference extends GET_MEDIA_TYPES implements Structure.ByReference {
+            public ByReference() {
+            }
```

Comment:
Don't leave this empty - add call to (implicit) _super()_ as this enables setting a breakpoint in the constructor